packages/ui/src/views/secrets/secrets-list/secrets-list-page.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC } from 'react'
+
+import { Button, ListActions, NoData, Pagination, SearchBox, Spacer } from '@/components'
+import { useRouterContext } from '@/context'
+import { useDebounceSearch } from '@/hooks'
+import { SandboxLayout } from '@/views'
+import { cn } from '@utils/cn'
+
+import { SecretList } from './secrets-list'
+import { SecretListPageProps } from './types'
+
+const SecretListPage: FC<SecretListPageProps> = ({
+  searchQuery,
+  setSearchQuery,
+  isError,
+  errorMessage,
+  useTranslationStore,
+  currentPage,
+  totalPages,
+  goToPage,
+  isLoading,
+  secrets,
+  ...props
+}) => {
+  const { t } = useTranslationStore()
+  const { navigate } = useRouterContext()
+
+  const { search: searchInput, handleSearchChange: handleInputChange } = useDebounceSearch({
+    handleChangeSearchValue: (val: string) => setSearchQuery(val.length ? val : undefined),
+    searchValue: searchQuery || ''
+  })
+
+  if (isError) {
```

Comment:
since loading state is handled in the list view.. maybe the error state should also be in the same place

Reply:
Sure, will clean up this as discussed in a follow up PR.